python/sglang/srt/server.py

```diff
@@ -475,6 +497,21 @@ def _set_envs_and_config(server_args: ServerArgs):
     os.environ["TORCH_NCCL_AVOID_RECORD_STREAMS"] = "1"
     os.environ["CUDA_DEVICE_MAX_CONNECTIONS"] = "4"
 
+    # Set prometheus multiprocess directory
+    # sglang uses prometheus multiprocess mode
+    # we need to set this before importing prometheus_client
+    # https://prometheus.github.io/client_python/multiprocess/
+    global prometheus_multiproc_dir
+    if "PROMETHEUS_MULTIPROC_DIR" in os.environ:
+        logger.debug(f"User set PROMETHEUS_MULTIPROC_DIR detected.")
+        prometheus_multiproc_dir = tempfile.TemporaryDirectory(
+            dir=os.environ["PROMETHEUS_MULTIPROC_DIR"]
+        )
+    else:
+        prometheus_multiproc_dir = tempfile.TemporaryDirectory()
+        os.environ["PROMETHEUS_MULTIPROC_DIR"] = prometheus_multiproc_dir.name
```

Comment:
one test case is failing for this.

1. start engine 1 => set env_var to tmp_dir_1
2. terminate engine1
3. start engine2 => env_var already exists but tmp_dir_1 is already cleaned up => thus error

we can get around this by not setting prom env when launching engine. the setting of prometheus should only happens in `launch_server` but not in launch_engine. maybe have a separate `_set_prometheus_env` function and use the func in `launch_server` only